modular_skyrat/modules/vore/code/datums/components/vore.dm

```diff
@@ -0,0 +1,157 @@
+	RegisterSignal(owner, COMSIG_PARENT_PREQDELETED, .proc/handle_delete)
+	update_bellies(TRUE)
+
+/datum/component/vore/Destroy()
+	UnregisterSignal(owner, COMSIG_PARENT_PREQDELETED)
+	handle_delete()
+	. = ..()
+
+/datum/component/vore/proc/update_current_slot()
+	if (!isnull(current_slot))
+		return current_slot
+	if (ishuman(owner))
+		var/mob/living/carbon/human/character = owner
+		current_slot = character.character_slot
+		if (!isnull(current_slot))
+			return current_slot
+	current_slot = owner.client?.prefs.default_slot
+	return current_slot
+
+/datum/component/vore/proc/handle_delete()
+	SIGNAL_HANDLER
+
+	owner.release_belly_contents()
+
+/datum/component/vore/proc/get_belly_contents(bellynum, ref=FALSE, living=FALSE, as_string=FALSE, ignored=null, full=FALSE)
+	if (bellynum < 1 || bellynum > bellies.len)
+		return
+	var/list/belly_contents = bellies[bellynum].get_belly_contents(ref, living, as_string, ignored, full)
+	return belly_contents
+
+/datum/component/vore/proc/update_bellies(set_ref=FALSE) //update everything
+	if (!owner.client?.prefs?.vr_prefs)
+		return
+	var/datum/vore_prefs/vore = owner.client.prefs.vr_prefs
+	vore_toggles = vore.vore_toggles
+	char_vars = vore.char_vars
+	selected_belly = vore.selected_belly
+	vore_enabled = vore.vore_enabled
+	for (var/bellynum in 1 to vore.bellies.len)
+		var/belly_ref = (set_ref ? bellynum : null)
+		if (bellynum > bellies.len)
+			var/obj/vbelly/belly = new(null, owner, vore.bellies[bellynum], belly_ref)
+			bellies += belly
+		else
+			var/obj/vbelly/belly = bellies[bellynum]
+			belly.set_data(vore.bellies[bellynum], belly_ref)
+
+/datum/component/vore/proc/update_belly(bellynum, data)
+	if (!bellynum || bellynum > bellies.len)
+		return
+	bellies[bellynum].set_data(data)
+
+/datum/component/vore/proc/remove_belly(bellynum)
+	if (!bellynum || bellynum > bellies.len)
+		return
+	var/obj/vbelly/belly = bellies[bellynum]
+	belly.mass_release_from_contents()
+	bellies.Cut(bellynum, bellynum+1)
+	qdel(belly)
+
+//change this to mob/living once/if you make simplemob vore a thing
+/mob/living/carbon/human/Login()
+	. = ..()
+	if (!. || !client)
+		return
+	update_vore_verbs()
+
+/mob/living/Login()
+	. = ..()
+	if (!. || !client)
+		return
+	add_verb(src, /mob/living/proc/OOC_Escape)
+
+/mob/living/proc/update_vore_verbs()
+	if (client.prefs?.vr_prefs?.vore_enabled)
+		var/datum/component/vore/vore = LoadComponent(/datum/component/vore)
+		client.prefs.vr_prefs.load_slotted_prefs() //make a way to save before this just in case has_unsaved is TRUE?
+		vore.update_bellies()
+		add_verb(src, /mob/living/verb/Ingest)
+	else
+		remove_verb(src, /mob/living/verb/Ingest)
+
+/mob/living/proc/OOC_Escape()
+	set name = "OOC Escape"
+	set category = "OOC"
+
+	while(istype(loc, /obj/vbelly))
+		var/obj/vbelly/belly = loc
+		forceMove(belly.drop_location())
+
+/mob/living/verb/Ingest(mob/living/prey in oview(1)) //this should be moved to some other way of doing things rather than a verb
```

Comment:
So, SS13 history lesson!

Back in the day, the game was a laggy piece of shit.

Part of this was, unironically, because of Verbs.

We used Verbs fucking everywhere with oview() and shit.

This makes shit lag because it calls that oview() a ***lot***.

Move this to action buttons, please.